crates/modelardb_server/src/storage/data_transfer.rs

```diff
@@ -190,14 +198,29 @@ impl DataTransfer {
         );
 
         // Merge the files and transfer them to the remote object store.
-        StorageEngine::merge_compressed_apache_parquet_files(
+        CompressedDataManager::merge_compressed_apache_parquet_files(
             &self.local_data_folder,
             &object_metas,
             &self.remote_data_folder,
             &format!("{COMPRESSED_DATA_FOLDER}/{table_name}/{column_index}"),
         )
         .await?;
 
+        // Delete the metadata for the transferred files from the metadata database.
+        let compressed_files_to_delete: Vec<Uuid> =
+            CompressedFile::object_metas_to_compressed_file_names(object_metas.clone())
+                .map_err(|error| ParquetError::General(error.to_string()))?;
+
+        self.metadata_manager
+            .replace_compressed_files(
+                table_name,
+                column_index as usize,
```

Comment:
For conversion that cannot fail it is better to use `into()` as it will then stop working if refactoring makes the conversion unsafe at some point in the future.

Reply:
Fixed for all cases of `column_index as usize`.